base/compiler/inferenceresult.jl

```diff
@@ -3,7 +3,7 @@
 function is_argtype_match(@nospecialize(given_argtype),
                           @nospecialize(cache_argtype),
                           overridden_by_const::Bool)
-    if isa(given_argtype, Const) || isa(given_argtype, PartialStruct)
+    if isa(given_argtype, Const) || isa(given_argtype, PartialStruct) || isa(given_argtype, PartialOpaque)
```

Comment:
Why would a PartialOpaque ever be equal with `OpaqueClosure`?

Reply:
You're right, it shouldn't be